cohttp-eio/src/cohttp_eio.mli

```diff
@@ -26,27 +26,48 @@ module Body : sig
   val pp_chunk : Format.formatter -> chunk -> unit
 end
 
-(** [Server] is a HTTP 1.1 server. *)
-module Server : sig
-  type middleware = handler -> handler
-  and handler = request -> response
-  and request = Http.Request.t * Eio.Buf_read.t
-  and response = Http.Response.t * Body.t
+module Request : sig
+  type t
+  (** An HTTP request. *)
+
+  val of_http :
+    reader:Eio.Buf_read.t ->
+    client_addr:Eio.Net.Sockaddr.stream ->
+    Http.Request.t -> t
+
+  val pp : t Fmt.t
 
-  (** {1 Request} *)
+  val client_addr : t -> Eio.Net.Sockaddr.stream
 
-  val read_fixed : request -> string
-  (** [read_fixed (request,reader)] is bytes of length [n] if "Content-Length"
-      header is a valid integer value [n] in [request]. [reader] is updated to
-      reflect that [n] bytes was read.
+  (** {2 Http.Request wrappers}
+
+      These provide direct access to the wrapped {!Http.Request.t}. *)
+
+  val http_request : t -> Http.Request.t
+  val has_body : t -> [ `No | `Unknown | `Yes ]
+  val headers : t -> Http.Header.t
+  val meth : t -> Http.Method.t
+  val scheme : t -> string option
+  val resource : t -> string
+  val version : t -> Http.Version.t
+  val encoding : t -> Http.Transfer.encoding
+  val is_keep_alive : t -> bool
+
+  val with_headers : Http.Header.t -> t -> t
```

Comment:
Our convention is that `t` comes first. You can label the header argument to make it work `|>`

Reply:
But as I've mentioned before, we don't need all these trivial functions. Keep the interface short and sweet.